api/plugins/action/deploy_target_config.py

```diff
@@ -90,35 +90,42 @@ def run(self, tmp=None, task_vars=None):
         return result
 
 def determine_required_updates(existing_configs, desired_configs):
-    addition_required = []  
-    deletion_required = [
-        config['id']
-        for config in existing_configs
-        if not any(
-            config['branches'] == desired['branches']
-            for desired in desired_configs
-        )
-    ]
-
+    addition_required = []
+    deletion_required = []
+
+    grouped_configs = {}
+    for config in existing_configs:
+        key = (config['branches'], config['pullrequests'], str(config['deployTarget']['id']), str(config['weight']))
+        if key not in grouped_configs:
+            grouped_configs[key] = []
+        grouped_configs[key].append(config)
+
+    #Identify duplicates and mark older ones for deletion
+    for key, configs in grouped_configs.items():
```

Comment:
Hey @rashed-k can you explain the need to do this? In the above iteration you know if you've added a deploy config based on the constructed key to the seen config list so you could mark them for delete at that time without this extra logic.